examples/research_projects/onnxruntime/text_to_image/train_text_to_image.py

```diff
@@ -28,30 +29,96 @@
 import transformers
 from accelerate import Accelerator
 from accelerate.logging import get_logger
+from accelerate.state import AcceleratorState
 from accelerate.utils import ProjectConfiguration, set_seed
 from datasets import load_dataset
 from huggingface_hub import create_repo, upload_folder
+from onnxruntime.training.optim.fp16_optimizer import FP16_Optimizer as ORT_FP16_Optimizer
 from onnxruntime.training.ortmodule import ORTModule
+from packaging import version
 from torchvision import transforms
 from tqdm.auto import tqdm
 from transformers import CLIPTextModel, CLIPTokenizer
+from transformers.utils import ContextManagers
 
 import diffusers
 from diffusers import AutoencoderKL, DDPMScheduler, StableDiffusionPipeline, UNet2DConditionModel
 from diffusers.optimization import get_scheduler
 from diffusers.training_utils import EMAModel
-from diffusers.utils import check_min_version
+from diffusers.utils import check_min_version, deprecate, is_wandb_available
 from diffusers.utils.import_utils import is_xformers_available
 
 
+if is_wandb_available():
+    import wandb
```

Comment:
might want to raise an error if there's a request to use wandb tracker and it's not available

Reply:
@bghira I'm trying to keep this example consistent with the non-optimized example under `diffusers/examples/text_to_image/train_text_to_image.py` 

Should this change be made there as well? Is it best to add that to this PR or open a new one adjusting both scripts?

Reply:
i'm not used to reviewing stuff or how code style works here, so please take my suggestions with a grain of salt, that's just how we do things in my day-to-day. if it's something we want to do, maybe i can take a stab at it later in follow-up PRs.

Reply:
sounds good, will resolve this thread and leave it to you to update the scripts later if necessary